Automattic/wp-calypso
# Fix `Activate this design` button for Marketplace themes

Closes https://github.com/Automattic/dotcom-forge/issues/3581

## Proposed Changes

While the AT transfer is running and the progress bar is shown, the progress is being polled from the back end. Calypso also stores the `is_wpcom_atomic` flag in state as part of the cached response from the `/rest/v1.2/sites/:siteId` endpoint.  When the AT transfer is complete, and the user is redirected to the thank you page, this data is stale and needs to be refreshed.

This PR adds a query to refresh site details on the thank you page.

## Testing Instructions



* with a SIMPLE site purchase the Business plan, or register a new test site from `/start/business`
* purchase a partner theme which has plugin dependencies (eg. `themes/makoney` or `themes/nokul`)
* after checkout, wait for the AT transfer to finish
* on the Thank You page, click the `Activate this theme` button
* check that the ActivationModal is shown (the eligibility modal before should **NOT** be shown)
   <img width="1892" alt="CleanShot 2023-09-01 at 13 42 08@2x" src="https://github.com/Automattic/wp-calypso/assets/11555574/d787dda7-3d66-4ead-9c45-67dc84ce8671">
* check that the theme is activated

## Pre-merge Checklist
